data_structures/linked_list/remove_nth_node_from_end_of_list.py

```diff
@@ -0,0 +1,123 @@
+"""Learn more about this algorithm: https://www.geeksforgeeks.org/delete-nth-node-from-the-end-of-the-given-linked-list/"""
+
+from __future__ import annotations
+
+
+class Node:
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.next = None
+
+
+class LinkedListExceptionError(Exception):
+    pass
+
+
+class LinkedList:
+    def __init__(self):  # type: ignore
+        self.head = None
+
+    # from middle_element_of_linked_list
+    def append(self, new_data: int) -> int:
+        """
+        >>> link = LinkedList()
+        >>> link.append(5)
+        5
+        >>> link.append(6)
+        6
+        """
+        new_node = Node(new_data)
+        new_node.next = self.head
+        self.head = new_node
+        if not self.head:
+            raise LinkedListExceptionError(
+                "Unexpected error: Head of linked list is None after append operation."
+            )
+        return self.head.data
+
+    def view(self) -> str:
+        """
+        >>> link = LinkedList()
+        >>> link.append(5)
+        >>> link.append(6)
+        >>> link.view()
+        '6->5'
+        """
+        ret = ""
+        temp = self.head
+        while temp is not None:
+            ret += str(temp.data)
+            ret += "->"
+            temp = temp.next
+        return ret[:-2]
+
+    def remove_nth_from_end(self, position_from_end: int) -> Node | None:
+        """
+        >>> link = LinkedList()
+        >>> link.remove_nth_from_end(3)
+        No element found.
+        >>> link.append(5)
+        5
+        >>> link.append(6)
+        6
+        >>> link.append(8)
+        8
+        >>> link.append(8)
+        8
+        >>> link.append(10)
+        10
+        >>> link.append(12)
+        12
+        >>> link.append(17)
+        17
+        >>> link.append(-25)
+        -25
+        >>> link.append(-20)
+        -20
+        >>> link.remove_nth_from_end(3)
+        5->6->8->8->10->12->-25->-20
+        >>>
+        """
+        # want to have two pointers, one at the start and the other k nodes forward
+        # We could complete this in one pass if we stored a self.size variable
+
+        if not self.head:
+            print("No element found.")
```

Comment:
CONTRIBUTING.md.  Algorithmic functions do not print().  Please raise an `IndexError`.  There are plenty of examples in this repo of raising Exceptions and creating tests for them.